internal/virtualcircuit/create.go

```diff
@@ -0,0 +1,157 @@
+package virtualcircuit
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	metal "github.com/equinix-labs/metal-go/metal/v1"
+	"github.com/spf13/cobra"
+)
+
+type vcParamBuilder interface {
+	SetName(string)
+	SetDescription(string)
+	SetTags([]string)
+}
+
+// compile-time assertion that parameter types implement vcParamBuilder
+var (
+	_ vcParamBuilder = (*metal.VlanVirtualCircuitCreateInput)(nil)
+	_ vcParamBuilder = (*metal.VrfVirtualCircuitCreateInput)(nil)
+	_ vcParamBuilder = (*metal.VlanVirtualCircuitUpdateInput)(nil)
+	_ vcParamBuilder = (*metal.VrfVirtualCircuitUpdateInput)(nil)
+)
+
+func setParams(p vcParamBuilder, name, description string, tags []string) {
+	if name > "" {
+		p.SetName(name)
+	}
+	if description > "" {
+		p.SetDescription(description)
+	}
+
+	if len(tags) > 0 {
+		p.SetTags(tags)
+	}
+}
+
+func (c *Client) Create() *cobra.Command {
+	var (
+		connectionID string
+		portID       string
+		name         string
+		description  string
+		projectID    string
+		vnid         string
+		vrf          string
+		subnet       string
+		customerIP   string
+		metalIP      string
+		md5          string
+		peerAsn      int
+		speed        int
+		nnVlan       int
+		tags         []string
+	)
+
+	createVirtualCricuitCmd := &cobra.Command{
+		Use:   `create  [-i connection_id] [-p port_id] [-P <project_id> ] -n <name> [-d <description>] [-v <vnid> ] [-V <vlan> ] [-s <speed> ] [-t <tags> ]`,
+		Short: "Creates an create-virtual-circuit for specific interconnection.",
+		Long:  "Creates an create-virtual-circuit for specific interconnection",
+		Example: `  # Creates a new virtual-circuit named "interconnection": 
+  metal virtual-circuit create [-i connection_id] [-p port_id] [-P <project_id> ] [-n <name>] [-d <description>] [-v <vnid> ] [-V <vlan> ] [-s <speed> ] [-t <tags> ]
+
+  metal vc create -i 81c9cb9e-b02f-4c73-9e04-06702f1380a0 -p 9c8f0c71-591d-42fe-9519-2f632761e2da -P b4673e33-0f48-4948-961a-c31d6edf64f8 -n test-inter  -d test-interconnection -v 15315810-2fda-48b8-b8cd-441ebab684b5 -V 1010 -s 100
+  
+  metal virtual-circuit create [-i connection_id] [-p port_id] [-P <project_id> ] [-n <name>] [-d <description>] [-R <vrfs>] [-M <md5sum>] [-a <peerASN>] [-S <subnet>] [-c <Customer IP>] [-m <Metal IP>]`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			var (
+				createInput metal.VirtualCircuitCreateInput
+			)
+			vrfFlag, _ := cmd.Flags().GetString("vrf")
+			if vrfFlag != "" {
+				nnVlanFlag, _ := cmd.Flags().GetInt("nnVlan")
```

Comment:
Instead of all the code in one function, lets break them down in separate functions. ex: `vcWithVrf()`, `vcWithVlan()` and `handleVcCreate()` etc